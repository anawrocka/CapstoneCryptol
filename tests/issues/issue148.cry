Salsa20_expansion : {a} (a >= 1, 2 >= a) => ([16*a][8], [16][8]) -> [64][8]
Salsa20_expansion(k, n) = z
  where
    [s0, s1, s2, s3] = if (`a == 2) then split("expand 32-byte k")
                                    else split("expand 16-byte k")
    k1 = take`{16}(k#zero)
    k2 = drop`{16}(zero#k)
    x = s0 # k1 # s1 # n # s2 # k2 # s3
    z = x

