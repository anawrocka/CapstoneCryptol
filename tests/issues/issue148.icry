:l issue148.cry
