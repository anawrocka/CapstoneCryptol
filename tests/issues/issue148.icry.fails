Known issue, see ticket #4.
