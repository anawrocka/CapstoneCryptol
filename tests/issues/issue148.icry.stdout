Should load and typecheck cleanly - no errors. Once it does, I'll put in a smoketest of reasonable output.
